crates/net/network-api/src/lib.rs

```diff
@@ -28,3 +32,31 @@ pub trait PeersInfo: Send + Sync {
     /// Returns the Ethereum Node Record of the node.
     fn local_node_record(&self) -> NodeRecord;
 }
+
+/// The status of the network being ran by the local node.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NetworkStatus {
+    /// The local node client name.
+    pub client_name: String,
+    /// The protocols the client is currently running.
+    pub protocols: BTreeMap<String, ProtocolInfo>,
```

Comment:
I think we can use all protocols as fields here, like

`eth_protocol_info: EthProtocolInfo`

Reply:
done in https://github.com/paradigmxyz/reth/pull/923/commits/5bb4fe0492add0091d6aa45d22e407a4df6ff7f2